### ISSUE TITLE
Widget Tests pass individually but fail if run together

### ISSUE THREAD

**Issue by KiraResari** [flutter/flutter#118433]
## Steps to Reproduce

1. Check out this branch: https://github.com/KiraResari/ceal-chronicler-f/tree/debug/widget-tests-fail-it-run-together
2. Run the widget tests in ceal-chronicler-f\test\characters\character_selection_view_test.dart individually
    * When run individually, the tests will pass
3. Run all the widget tests in ceal-chronicler-f\test\characters\character_selection_view_test.dart at once

**Expected results:** 
* The tests run together should exhibit the same behavior as when run individually.
* In this case that means they should all pass.

**Actual results:**
* The tests run together should exhibit a different behavior as when run individually.
* In this case, the first test passes as normal, and all subsequent tests fail with the following error:
```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching node in the widget tree
  Actual: _TextFinder:<zero widgets with text "Unnamed Character" (ignoring offstage widgets)>
   Which: means none were found but one was expected

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///C:/projects/ceal-chronicler-f/test/characters/character_selection_view_test.dart:28:7)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///C:/projects/ceal-chronicler-f/test/characters/character_selection_view_test.dart line 28
The test description was:
  Clicking Add Button should add a character
════════════════════════════════════════════════════════════════════════════════════════════════════

Test failed. See exception logs above.
The test description was: Clicking Add Button should add a character
```

<details>
<summary>Code sample</summary>

**Dependencies**

```
  provider: ^6.0.0
  get_it: ^7.2.0
  event_bus: ^2.0.0
```

**Code**

```
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:event_bus/event_bus.dart';
import 'package:get_it/get_it.dart';

class CealChroniclerF extends StatelessWidget {
  const CealChroniclerF({super.key});

  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => AppState(),
      child: const MaterialApp(
        title: "Ceal Chronicler f",
        home: MainView(),
      ),
    );
  }
}

class MainView extends StatelessWidget {
  const MainView({super.key});

  @override
  Widget build(BuildContext context) {
    var appState = context.watch<AppState>();

    return Material(child: appState.activeView);
  }
}

class AppState extends ChangeNotifier {
  Widget activeView = const TitleView();

  final _characterSelectionView = const CharacterSelectionView();

  AppState() {
    var eventBus = getIt.get<EventBus>();
    eventBus.on<OpenCharacterSelectionViewEvent>().listen(
      (event) {
        _onOpenCharacterSelectionViewEvent();
      },
    );
  }

  void _onOpenCharacterSelectionViewEvent() {
    activeView = _characterSelectionView;
    notifyListeners();
  }
}

class TitleView extends StatelessWidget {
  static const String startButtonText = "Go to Character Selection Screen";

  const TitleView({super.key});

  @override
  Widget build(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [_buildStartButton()],
    );
  }

  ElevatedButton _buildStartButton() {
    var eventBus = getIt.get<EventBus>();

    return ElevatedButton(
      onPressed: () {
        eventBus.fire(OpenCharacterSelectionViewEvent());
      },
      child: const Text(startButtonText),
    );
  }
}

class CharacterSelectionView extends StatefulWidget {
  static const String addCharacterButtonText = "Add Character";
  static final model = CharacterSelectionModel();

  const CharacterSelectionView({super.key});

  @override
  State<CharacterSelectionView> createState() => _CharacterSelectionViewState();
}

class _CharacterSelectionViewState extends State<CharacterSelectionView> {
  List<Character> characters = CharacterSelectionView.model.characters;
  final _eventBus = getIt.get<EventBus>();

  _CharacterSelectionViewState() {
    _eventBus.on<UpdateCharacterSelectionViewEvent>().listen((event) {
      _updateCharacterSelectionView();
    });
  }

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(8.0),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          _buildCharactersColumn(context),
          _buildAddButton(context),
        ],
      ),
    );
  }

  Column _buildCharactersColumn(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: _buildCharacterButtons(context),
    );
  }

  List<Widget> _buildCharacterButtons(BuildContext context) {
    List<Widget> characterButtons = [];
    for (var character in characters) {
      var button = _buildCharacterButton(character, context);
      characterButtons.add(button);
    }
    return characterButtons;
  }

  Widget _buildCharacterButton(Character character, BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(5),
      child: ElevatedButton(
        onPressed: () => _onCharacterButtonPressed(character),
        child: Text(character.name),
      ),
    );
  }

  void _onCharacterButtonPressed(Character character) {}

  Widget _buildAddButton(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(8.0),
      child: ElevatedButton(
        onPressed: () => _onAddButtonPressed(),
        child: const Text(CharacterSelectionView.addCharacterButtonText),
      ),
    );
  }

  void _onAddButtonPressed() {
    _eventBus.fire(AddCharacterEvent());
  }

  void _updateCharacterSelectionView() {
    setState(
      () {
        characters = CharacterSelectionView.model.characters;
      },
    );
  }
}

class CharacterSelectionModel {
  final List<Character> _characters = [];
  final _eventBus = getIt.get<EventBus>();

  CharacterSelectionModel() {
    _eventBus.on<AddCharacterEvent>().listen(
      (event) {
        addCharacter();
      },
    );
  }

  List<Character> get characters {
    return _characters;
  }

  void addCharacter() {
    var character = Character();
    _characters.add(character);
    _eventBus.fire(UpdateCharacterSelectionViewEvent());
  }
}

class Character {
  static const defaultName = "Unnamed Character";

  String name;

  Character([name]) : name = name ?? defaultName;
}

class AddCharacterEvent {}

class UpdateCharacterSelectionViewEvent {}

class OpenCharacterSelectionViewEvent {}

final getIt = GetIt.instance;

void initializeGetItContext() {
  getIt.registerSingleton<EventBus>(EventBus());
}

void main() {
  setUp(() => initializeGetItContext());
  tearDown(() => getIt.reset());

  testWidgets(
    "Character Selection View should have Add button",
    (WidgetTester tester) async {
      await firstTest(tester);
    },
  );

  testWidgets(
    "Clicking Add Button should add a character",
    (WidgetTester tester) async {
      await secondTest(tester);
    },
  );
}

Future<void> firstTest(WidgetTester tester) async {
  await navigateToCharacterSelectionView(tester);

  expect(
      find.text(CharacterSelectionView.addCharacterButtonText), findsOneWidget);
}

Future<void> secondTest(WidgetTester tester) async {
  await navigateToCharacterSelectionViewAndAddCharacter(tester);

  expect(find.text(Character.defaultName), findsOneWidget);
}

Future<void> navigateToCharacterSelectionViewAndAddCharacter(
    WidgetTester tester) async {
  await navigateToCharacterSelectionView(tester);

  await tester.tap(
      find.bySemanticsLabel(CharacterSelectionView.addCharacterButtonText));
  await tester.pump();
}

Future<void> navigateToCharacterSelectionView(WidgetTester tester) async {
  await tester.pumpWidget(const CealChroniclerF());

  await tester.tap(find.bySemanticsLabel(TitleView.startButtonText));
  await tester.pump();
}
```

</details>

<details>
  <summary>Logs</summary>

```
C:\projects\ceal-chronicler-f>flutter test --verbose
[  +66 ms] executing: [C:\sdks\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +105 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 135454af32477f815a7525073027a3ff9eff1bfd
[        ] executing: [C:\sdks\flutter/] git tag --points-at 135454af32477f815a7525073027a3ff9eff1bfd
[  +94 ms] Exit code 0 from: git tag --points-at 135454af32477f815a7525073027a3ff9eff1bfd
[        ] 3.3.10
[  +56 ms] executing: [C:\sdks\flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +80 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/stable
[        ] executing: [C:\sdks\flutter/] git ls-remote --get-url origin
[  +66 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +96 ms] executing: [C:\sdks\flutter/] git rev-parse --abbrev-ref HEAD
[  +75 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[  +75 ms] Found 3 files which will be executed as Widget Tests.
[   +6 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +42 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +35 ms] Skipping pub get: version match.
[ +128 ms] Generating
C:\projects\ceal-chronicler-f\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[ +129 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more
information see #56466.
[   +9 ms] running test package with arguments: [-r, compact, --timeout, 30s, --concurrency=6, --chain-stack-traces, --,
C:\projects\ceal-chronicler-f\test\characters\character_selection_model_test.dart,
C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart,
C:\projects\ceal-chronicler-f\test\widget_test.dart]
00:00 +0: loading C:\projects\ceal-chronicler-f\test\characters\character_selection_model_test.dart                    [ +214 ms] test 0: starting test C:\projects\ceal-chronicler-f\test\characters\character_selection_model_test.dart
[   +9 ms] Stopping scan for flutter_test_config.dart; found project root at C:\projects\ceal-chronicler-f
[   +5 ms] Compiler will use the following file as its incremental dill file:
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_compiler.86b281ef\output.dill
[   +1 ms] Listening to compiler controller...
[  +13 ms] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.d049785b/listener.dart
[  +46 ms] C:\sdks\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev
C:\sdks\flutter\bin\cache\dart-sdk\bin\snapshots\frontend_server.dart.snapshot --sdk-root
C:\sdks\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --no-print-incremental-dependencies
--target=flutter --debugger-module-names --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_compiler.86b281ef\output.dill --packages
C:\projects\ceal-chronicler-f\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false
--enable-asserts --track-widget-creation --initialize-from-dill
C:\projects\ceal-chronicler-f\build\test_cache\build\c075001b96339384a97db4862b8ab8db.cache.dill.track.dill
--verbosity=error
[   +3 ms] test 1: starting test C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart
[   +2 ms] Stopping scan for flutter_test_config.dart; found project root at C:\projects\ceal-chronicler-f
[   +1 ms] test 2: starting test C:\projects\ceal-chronicler-f\test\widget_test.dart
[   +2 ms] Stopping scan for flutter_test_config.dart; found project root at C:\projects\ceal-chronicler-f
[  +96 ms] <- compile
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.d049785b/listener.dart
00:01 +0: loading C:\projects\ceal-chronicler-f\test\characters\character_selection_model_test.dart                    [+1120 ms] <- accept
[   +1 ms] <- reset
[        ] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.d049785b/listener.dart took
1275ms
[        ] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.82f7b91b/listener.dart
[   +1 ms] test 0: starting test device
[   +3 ms] test 0: awaiting connection to test device
[  +15 ms] test 0: Observatory uri is not available
[   +4 ms] test 0: test harness socket server is running at port:55083
[   +3 ms] Using this directory for fonts configuration:
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_fonts.8e5dd7e8
[   +5 ms] test 0: Starting flutter_tester process with
command=[C:\sdks\flutter\bin\cache\artifacts\engine\windows-x64\flutter_tester.exe, --disable-observatory,
--enable-checked-mode, --verify-entry-points, --enable-software-rendering, --skia-deterministic-rendering,
--enable-dart-profiling, --non-interactive, --use-test-fonts, --disable-asset-fonts,
--packages=C:\projects\ceal-chronicler-f\.dart_tool\package_config.json,
--flutter-assets-dir=C:\projects\ceal-chronicler-f\build\unit_test_assets,
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_listener.d049785b\listener.dart.dill],
environment={FLUTTER_TEST: true, FONTCONFIG_FILE:
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_fonts.8e5dd7e8/fonts.conf, SERVER_PORT: 55083,
APP_NAME: ceal_chronicler_f, UNIT_TEST_ASSETS: C:\projects\ceal-chronicler-f\build\unit_test_assets}
[   +4 ms] <- recompile
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.82f7b91b/listener.dart
5ea61247-bd6e-4fd1-9661-734b87076e38
[   +1 ms]
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.82f7b91b/listener.dart
[        ] file:///C:/projects/ceal-chronicler-f/.dart_tool/flutter_build/dart_plugin_registrant.dart
[        ] <- 5ea61247-bd6e-4fd1-9661-734b87076e38
[  +59 ms] test 0: Started flutter_tester process at pid 1252
[ +184 ms] <- accept
[   +1 ms] <- reset
[   +1 ms] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.82f7b91b/listener.dart took
284ms
[   +1 ms] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.df6fdd96/listener.dart
[   +1 ms] test 1: starting test device
[        ] test 1: awaiting connection to test device
[  +14 ms] test 1: Observatory uri is not available
[   +1 ms] test 1: test harness socket server is running at port:55084
[        ] test 1: Starting flutter_tester process with
command=[C:\sdks\flutter\bin\cache\artifacts\engine\windows-x64\flutter_tester.exe, --disable-observatory,
--enable-checked-mode, --verify-entry-points, --enable-software-rendering, --skia-deterministic-rendering,
--enable-dart-profiling, --non-interactive, --use-test-fonts, --disable-asset-fonts,
--packages=C:\projects\ceal-chronicler-f\.dart_tool\package_config.json,
--flutter-assets-dir=C:\projects\ceal-chronicler-f\build\unit_test_assets,
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_listener.82f7b91b\listener.dart.dill],
environment={FLUTTER_TEST: true, FONTCONFIG_FILE:
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_fonts.8e5dd7e8/fonts.conf, SERVER_PORT: 55084,
APP_NAME: ceal_chronicler_f, UNIT_TEST_ASSETS: C:\projects\ceal-chronicler-f\build\unit_test_assets}
[   +1 ms] <- recompile
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.df6fdd96/listener.dart
d291bd9c-623b-42e2-8c48-b5f8f29b87f0
[        ]
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.df6fdd96/listener.dart
[        ] file:///C:/projects/ceal-chronicler-f/.dart_tool/flutter_build/dart_plugin_registrant.dart
[        ] <- d291bd9c-623b-42e2-8c48-b5f8f29b87f0
[  +55 ms] test 1: Started flutter_tester process at pid 4260
[ +161 ms] <- accept
[   +1 ms] <- reset
[        ] Compiling
file:///C:/Users/IV0308/AppData/Local/Temp/flutter_tools.d3af3251/flutter_test_listener.df6fdd96/listener.dart took
241ms
[        ] test 2: starting test device
[   +1 ms] test 2: awaiting connection to test device
[   +1 ms] test 2: Observatory uri is not available
[   +1 ms] test 2: test harness socket server is running at port:55085
[        ] test 2: Starting flutter_tester process with
command=[C:\sdks\flutter\bin\cache\artifacts\engine\windows-x64\flutter_tester.exe, --disable-observatory,
--enable-checked-mode, --verify-entry-points, --enable-software-rendering, --skia-deterministic-rendering,
--enable-dart-profiling, --non-interactive, --use-test-fonts, --disable-asset-fonts,
--packages=C:\projects\ceal-chronicler-f\.dart_tool\package_config.json,
--flutter-assets-dir=C:\projects\ceal-chronicler-f\build\unit_test_assets,
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_listener.df6fdd96\listener.dart.dill],
environment={FLUTTER_TEST: true, FONTCONFIG_FILE:
C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_fonts.8e5dd7e8/fonts.conf, SERVER_PORT: 55085,
APP_NAME: ceal_chronicler_f, UNIT_TEST_ASSETS: C:\projects\ceal-chronicler-f\build\unit_test_assets}
[  +66 ms] test 2: Started flutter_tester process at pid 3624
00:02 +0: loading C:\projects\ceal-chronicler-f\test\characters\character_selection_model_test.dart                    [ +244 ms] test 0: connected to test device, now awaiting test result
[   +1 ms] test 0: Waiting for test harness or tests to finish
00:02 +1: ... AddCharacter should increase Character Count by one                                                      [ +113 ms] test 0: Test harness is no longer needed by test process
[   +1 ms] test 0: finished
[   +3 ms] test 0: cleaning up...
[        ] test 0: ensuring test device is terminated.
[        ] test 0: Terminating flutter_tester process
[        ] test 0: Shutting down test harness socket server
[        ] test 0: Test process is no longer needed by test harness
[  +13 ms] test 0: flutter_tester process at pid 1252 exited with code=-1
[   +1 ms] test 0: deleting temporary directory
[   +4 ms] test 0: finished
[ +142 ms] test 1: connected to test device, now awaiting test result
[   +1 ms] test 1: Waiting for test harness or tests to finish
00:02 +1: ... Character Selection View should have Add button                                                          [ +192 ms] test 2: connected to test device, now awaiting test result
[   +1 ms] test 2: Waiting for test harness or tests to finish
00:03 +3: C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart: Clicking Add Button should add a character
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching node in the widget tree
  Actual: _TextFinder:<zero widgets with text "Unnamed Character" (ignoring offstage widgets)>
   Which: means none were found but one was expected

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///C:/projects/ceal-chronicler-f/test/characters/character_selection_view_test.dart:28:7)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///C:/projects/ceal-chronicler-f/test/characters/character_selection_view_test.dart line 28
The test description was:
  Clicking Add Button should add a character
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +3 -1: C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart: Clicking Add Button should add a character [E]
  Test failed. See exception logs above.
  The test description was: Clicking Add Button should add a character


To run this test again: C:\sdks\flutter\bin\cache\dart-sdk\bin\dart.exe test C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart -p vm --plain-name 'Clicking Add Button should add a character'
00:03 +4 -1: C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart: Clicking Character Button should open character view
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown running a test:
The finder "zero widgets with element matching predicate (Closure: (Element) => bool) (ignoring
offstage widgets)" (used in a call to "tap()") could not find any matching widgets.

When the exception was thrown, this was the stack:
#0      WidgetController._getElementPoint (package:flutter_test/src/controller.dart:907:7)
#1      WidgetController.getCenter (package:flutter_test/src/controller.dart:846:12)
#2      WidgetController.tap (package:flutter_test/src/controller.dart:274:18)
#3      main.<anonymous closure> (file:///C:/projects/ceal-chronicler-f/test/characters/character_selection_view_test.dart:37:20)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

The test description was:
  Clicking Character Button should open character view
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +4 -2: C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart: Clicking Character Button should open character view [E]
  Test failed. See exception logs above.
  The test description was: Clicking Character Button should open character view


To run this test again: C:\sdks\flutter\bin\cache\dart-sdk\bin\dart.exe test C:\projects\ceal-chronicler-f\test\characters\character_selection_view_test.dart -p vm --plain-name 'Clicking Character Button should open character view'
[ +927 ms] test 1: Test harness is no longer needed by test process
[        ] test 1: finished
[        ] test 1: cleaning up...
[        ] test 1: ensuring test device is terminated.
[        ] test 1: Terminating flutter_tester process
[        ] test 1: Shutting down test harness socket server
[        ] test 1: Test process is no longer needed by test harness
[  +21 ms] test 1: flutter_tester process at pid 4260 exited with code=-1
[   +1 ms] test 1: deleting temporary directory
[   +5 ms] test 1: finished
00:03 +5 -2: C:\projects\ceal-chronicler-f\test\widget_test.dart: Navigating to Character Selection should work        [   +2 ms] test 2: Test harness is no longer needed by test process
[        ] test 2: finished
[        ] test 2: cleaning up...
[        ] test 2: ensuring test device is terminated.
[        ] test 2: Terminating flutter_tester process
[        ] test 2: Shutting down test harness socket server
[        ] test 2: Test process is no longer needed by test harness
[  +17 ms] test 2: flutter_tester process at pid 3624 exited with code=-1
[        ] test 2: deleting temporary directory
[   +3 ms] test 2: finished
00:03 +5 -2: Some tests failed.
[   +8 ms] Deleting C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_compiler.86b281ef...
[   +4 ms] killing pid 10056
[  +41 ms] Deleting C:\Users\IV0308\AppData\Local\Temp\flutter_tools.d3af3251\flutter_test_fonts.8e5dd7e8...
[   +3 ms] test package returned with exit code 1
[   +7 ms] "flutter test" took 4.279ms.
[   +7 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      TestCommand.runCommand (package:flutter_tools/src/commands/test.dart:467:7)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command.dart:1209:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>


[  +69 ms] ensureAnalyticsSent: 66ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

```
C:\projects\ceal-chronicler-f>flutter analyze
Analyzing ceal-chronicler-f...
No issues found! (ran in 1.8s)
```

```
C:\projects\ceal-chronicler-f>flutter doctor -v
[√] Flutter (Channel stable, 3.3.10, on Microsoft Windows [Version 10.0.19044.2364], locale de-DE)
    • Flutter version 3.3.10 on channel stable at C:\sdks\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (4 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at C:\Users\IV0308\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: C:\Programs\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33205.214
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Programs\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.2364]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 107.0.5304.88
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 108.0.1462.76

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

I am experiencing that some of my widget tests run individually, but not as a group, and am at a loss for why. Searches on StackOverflow have revealed that others are also experiencing this issue, and no working solution has been found to date:

* https://stackoverflow.com/questions/66598442/flutter-widget-tests-do-not-pass-when-run-together
* https://stackoverflow.com/questions/66896834/flutter-widget-test-failed-when-running-together
* https://stackoverflow.com/questions/68997984/why-do-some-flutter-widget-tests-fail-if-executed-together-but-pass-individually

Is there something I am doing wrong here? Or is this an issue with Flutter that I can do nothing about? At any rate, I am at a complete loss here.

My basic understanding of a testing framework is that it should instantiate the tests in a way that they do not interact with one another. Assuming that holds true, then I could only imagine that some interaction happens through the getIt context. This is somehow unexpected since I create a new context before every test and tear it down after every test. I have no indication that the tests run in parallel, but parallel execution of the tests would at least offer basic grounds for a potential explanation of this behavior. However, even it that were to be true, I would _still_ have no idea of what to do about that.

**Comment by exaby73** [comment]
Hello @KiraResari. Thank you for the issue report. I can see as a code sample, you provided an application link. Unfortunately, I would require a [complete and minimal, reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) to verify the issue. Could you isolate the issue in a new Flutter project and provide the code sample(s)?

**Comment by KiraResari** [comment]
Okay, I added a code sample now. Not sure how much sense this makes since that's more or less just the entire project dumped into one file, but at least it reproduces the issue when run.

**Comment by exaby73** [comment]
Hey @KiraResari. What I meant by isolating the issue is to produce a minimal example. The code should have minimal code to reproduce the issue, and any code that can be removed, should be removed if it doesn't help reproduce the issue. Please read [complete and minimal, reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) for more information on such a code sample

**Comment by KiraResari** [comment]
Very well, I now spent more time on this, and I managed to get the code sample down to what I believe is the minimum possible size. I managed to get some dependencies out, but GetIt and the EventBus are just too essential and deep-rooted. Also, I have a strong suspicion that since this feels like a threading issue, either one of those in conjunction with the Flutter Widget Testing framework is a likely culprit (everything works fine with non-widget tests though). The code sample is now shorter than the logs, which is as short as I can get it, and now contains a minimum of two tests, which pass when run separately, but fail if run together. At the beginning I was easily able to remove a large amount of unneeded clutter, but near the end I really started struggling hunting for individual lines that I could still remove without detrimentally affecting the legibility of the code sample. So this is as minimal a sample as I can provide. I hope this is sufficient for you to investigate and hopefully resolve this issue.

**Comment by KiraResari** [comment]
Hmm, I am not happy with that. For starters, I still have no idea what to do about this then. And then, it definitely does feel like a framework issue that the order in which the tests are run affect the results. Also, I don't think that what you said is quite applicable here, because first step of each of these tests is to click on the Start button of the title screen. If the app did not correctly start over, then that button would not be there, and the test would fail right there and then. Instead, they only fail at the very last stage, after successfully navigating through the app from the title view and adding a new character, when checking if that character is there. So can you please have a more thorough look into this?

**Comment by exaby73** [comment]
Could you possibly create a sample what does not have any 3rd party libraries (i.e provider, event_bus, etc) that still reproduces the issue? This could be an issue that the test is finishing before the asynchronous nature of the way you're handling events is done. Therefore, if you can provide a sample that reproduces this issue without event_bus, this would be helpful in figuring if the issue is indeed with Flutter and/or flutter_test
